### ISSUE TITLE
Update the canonical URL if present

### ISSUE THREAD

**Issue by bglw** [CloudCannon/pagebreak#3]
If a page contains a canonical URL, it should be updated to point the paginated page to itself.

If `index.html` contains:
```html
<link rel="canonical" href="https://example.com/">
```

Then `page/2/index.html` should contain:
```html
<link rel="canonical" href="https://example.com/page/2/">
```

**Comment by bglw** [comment]
Skipped test added in the above commit

**Comment by bglw** [comment]
Fixed in #17
